pytheus/metrics.py

```diff
@@ -48,15 +53,21 @@ def _validate_labels(self, labels: Sequence[str]):
 
 # class or function based creation ?
 class Metric:
-
     def __init__(
         self,
         collector: MetricCollector,
         labels: Labels | None = None,
+        backend_class: Optional[Backend] = None,
+        backend_config: Optional[BackendConfig] = None,
     ) -> None:
         self._collector = collector
         self._labels = labels
         self._can_observe = self._check_can_observe()
+        if backend_class is not None and backend_config is not None:
+            self._backend = backend_class(backend_config)
+        else:
+            from pytheus.backends import DEFAULT_BACKEND_CLASS, DEFAULT_BACKEND_CONFIG
```

Comment:
Not a very elegant way of caching these, but it's very simple and works? Any other ideas?

Reply:
we can consider renaming `DEFAULT_BACKEND_CLASS` to a generic `BACKEND_CLASS` that will be overridden when loading from env.
Then we can use that to instantiate backend instances.

Reply:
89873829500450ebc594a994fc28443e16ed6a4b